### DISCUSSION TITLE
Validate Resource Route

### DISCUSSION THREAD

**Discussion by hsharma1996** [adonisjs/core#3276] [Help]
How to validate resource route in adonis ? Any clean practise ? 
Since there is no .validate function, 
I guess I have to create a middleware and import validation schemas into that and apply them based on request method.

**Comment by RomainLanz** [comment]
Hey @hsharma1996! 👋🏻 

The `.validate()` method is for AdonisJS 4.
In version 5, you should validate your payload inside your controller handler.
